### PR TITLE
Refactor Vision to use Pose and Twist

### PR DESCRIPTION
**Pull request by Tdogb**
## Description
Refactoring of Kalman filter to use Pose and Twist

## Associated Issue
Issue #1317

### CODE REVIEW

**Review by JNeiger — APPROVED** [review]

File: soccer/vision/robot/KalmanRobot.cpp [hunk 1]
```diff
@@ -23,20 +23,18 @@ KalmanRobot::KalmanRobot(unsigned int cameraID, RJ::Time creationTime,
       unwrapThetaCtr(0), robotID(initMeasurement.getRobotID()),
       previousMeasurements(*VisionFilterConfig::slow_kick_detector_history_length) {
 
-    Geometry2d::Point initPos = initMeasurement.getPos();
-    double initTheta          = initMeasurement.getTheta();
-    Geometry2d::Point initVel = Geometry2d::Point(0, 0);
-    double initOmega          = 0;
+    Geometry2d::Pose initPose(initMeasurement.getPos().x(), initMeasurement.getPos().y(), initMeasurement.getTheta());
```
Comment (kylestach) [inline soccer/vision/robot/KalmanRobot.cpp]: `CameraRobot` should use `Pose` internally.

File: soccer/vision/robot/KalmanRobot.cpp [hunk 2]
```diff
@@ -23,20 +23,18 @@ KalmanRobot::KalmanRobot(unsigned int cameraID, RJ::Time creationTime,
...
+    Geometry2d::Twist initTwist(0,0,0);
 
     if (previousWorldRobot.getIsValid()) {
-        initVel = previousWorldRobot.getVel();
-        initOmega = previousWorldRobot.getOmega();
+        initTwist.linear() = previousWorldRobot.getVel();
```
Comment (kylestach) [inline soccer/vision/robot/KalmanRobot.cpp:29]: Most likely, `WorldRobot` should use `Pose/Twist` classes too.

File: soccer/vision/filter/KalmanFilter3D.cpp [hunk 3]
```diff
@@ -104,11 +104,10 @@ KalmanFilter3D::KalmanFilter3D(Geometry2d::Point initPos, double initTheta,
...
-    z_k << observationPos.x(),
-           observationPos.y(),
-           observationTheta;
+void KalmanFilter3D::predictWithUpdate(Geometry2d::Pose observation) {
+    z_k << observation.position().x(),
+           observation.position().y(),
+           observation.heading();
 
     KalmanFilter::predictWithUpdate();
 }
```
Comment (kylestach) [inline soccer/vision/filter/KalmanFilter3D.cpp:108]: KalmanFilter3D should have a way of getting a Pose and a Twist.

**Review by kylestach — CHANGES_REQUESTED** [review]
Thanks for taking the initiative on this one! The idea behind this is a lot more extensive, though: every time we have a Point for position or linear velocity combined with a double for heading or angular velocity, that should be replaced with a Pose or a Twist, respectively. I can walk you through this more on Tuesday's meeting.

Feel free to ping me with any questions on this.

**Review by kylestach — APPROVED** [review]
LGTM

**Pull request merged**